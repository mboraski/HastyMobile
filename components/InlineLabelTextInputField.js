import React from 'react';
import { View, Text, TextInput, StyleSheet } from 'react-native';
import { Field } from 'redux-form';

import Color from '../constants/Color';
import { emY } from '../utils/em';

const renderInput = ({ input: { onChange, ...restInput } }) => (
    <TextInput style={styles.textInput} onChangeText={onChange} {...restInput} />
);

const InlineLabelTextInputField = ({ label, ...props }) => (
    <View style={styles.formInputGroup}>
        <Text style={styles.label}>{label}</Text>
        <View style={styles.textInputContainer}>
            <Field component={renderInput} style={styles.textInput} {...props} />
        </View>
    </View>
);

const styles = StyleSheet.create({
    formInputGroup: {
        flexDirection: 'row',
        paddingHorizontal: 15,
        backgroundColor: Color.GREY_100,
        borderColor: Color.GREY_300,
        borderBottomWidth: StyleSheet.hairlineWidth,
        alignItems: 'center'
    },
    label: {
        color: '#000',
        fontSize: emY(1),
        paddingVertical: emY(0.875),
        marginRight: 15
    },
    textInputContainer: {
        flex: 1
    },
    textInput: {
        flex: 1,
        fontSize: emY(1),
        textAlign: 'right'
    }
});

export default InlineLabelTextInputField;
