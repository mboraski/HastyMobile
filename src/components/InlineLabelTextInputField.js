import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Field } from 'redux-form';

import Text from './Text';
import Color from '../constants/Color';
import { emY } from '../utils/em';

export const InlineLabelTextInput = ({
    input: { onChange, ...restInput },
    meta: { touched, invalid, error },
    label,
    containerStyle,
    labelStyle,
    inputContainerStyle,
    style,
    ...props
}) => (
    <View style={[styles.container, containerStyle]}>
        <View style={styles.formInputGroup}>
            <Text style={[styles.label, labelStyle, touched && invalid && styles.labelInvalid]}>
                {label}
            </Text>
            <View style={[styles.textInputContainer, inputContainerStyle]}>
                <TextInput
                    style={[styles.textInput, style, touched && invalid && styles.textInputInvalid]}
                    onChangeText={onChange}
                    {...restInput}
                    {...props}
                />
            </View>
        </View>
        {touched && error ? <Text style={styles.error}>{error}</Text> : null}
    </View>
);

const InlineLabelTextInputField = ({ textInputName, ...props }) => (
    <Field name={textInputName} component={InlineLabelTextInput} {...props} />
);

const styles = StyleSheet.create({
    container: {
        backgroundColor: Color.GREY_100
    },
    formInputGroup: {
        paddingHorizontal: 15,
        borderColor: Color.GREY_300,
        borderBottomWidth: StyleSheet.hairlineWidth,
        flexDirection: 'row',
        alignItems: 'center',
        marginBottom: 5
    },
    label: {
        color: '#000',
        fontSize: emY(1),
        paddingVertical: emY(0.875),
        marginRight: 15
    },
    labelInvalid: {
        color: Color.RED_500
    },
    textInputContainer: {
        flex: 1
    },
    textInput: {
        flex: 1,
        fontSize: emY(1),
        textAlign: 'right',
        fontFamily: 'Arial'
    },
    textInputInvalid: {
        color: Color.RED_500
    },
    error: {
        paddingHorizontal: 15,
        fontSize: emY(0.8),
        color: Color.RED_500,
        marginBottom: 10
    }
});

export default InlineLabelTextInputField;
