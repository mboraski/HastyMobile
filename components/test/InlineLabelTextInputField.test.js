import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import InlineLabelTextInputField from '../InlineLabelTextInputField';

describe('InlineLabelTextInputField', () => {
    it('renders correctly', () => {
        const render = shallow(<InlineLabelTextInputField textInputName={'Test_Input_Name'} />);
        expect(render).toMatchSnapshot();
    });
});
