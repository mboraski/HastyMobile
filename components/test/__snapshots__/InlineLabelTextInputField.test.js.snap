// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLabelTextInputField component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Array [
            Object {
              "backgroundColor": "#F4F4F4",
              "borderBottomWidth": 0.5,
              "borderColor": "#cccccc",
              "paddingHorizontal": 15,
            },
            undefined,
          ]
    }
>
    <View
        style={
            Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
        }
    >
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Array [
                    Object {
                      "color": "#000",
                      "fontSize": 16,
                      "marginRight": 15,
                      "paddingVertical": 14,
                    },
                    undefined,
                    false,
                  ]
            }
        >
            label
        </Text>
        <View
            style={
                Array [
                    Object {
                      "flex": 1,
                    },
                    undefined,
                  ]
            }
        >
            <TextInput
                onChangeText={undefined}
                style={
                    Array [
                        Object {
                          "flex": 1,
                          "fontSize": 16,
                          "textAlign": "right",
                        },
                        undefined,
                        false,
                      ]
                }
            />
        </View>
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Array [
                  Object {
                    "backgroundColor": "#F4F4F4",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#cccccc",
                    "paddingHorizontal": 15,
                  },
                  undefined,
                ]
      }
>
      <View
            style={
                  Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
            }
      >
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Array [
                              Object {
                                "color": "#000",
                                "fontSize": 16,
                                "marginRight": 15,
                                "paddingVertical": 14,
                              },
                              undefined,
                              false,
                            ]
                  }
            >
                  label
            </Text>
            <View
                  style={
                        Array [
                              Object {
                                "flex": 1,
                              },
                              undefined,
                            ]
                  }
            >
                  <TextInput
                        onChangeText={undefined}
                        style={
                              Array [
                                    Object {
                                      "flex": 1,
                                      "fontSize": 16,
                                      "textAlign": "right",
                                    },
                                    undefined,
                                    false,
                                  ]
                        }
                  />
            </View>
      </View>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <InlineLabelTextInput
        input={Object {}}
        label="label"
        meta={
                Object {
                        "error": null,
                        "invalid": false,
                        "touched": false,
                      }
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "input": Object {},
          "label": "label",
          "meta": Object {
            "error": null,
            "invalid": false,
            "touched": false,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Array [
                              Object {
                                "backgroundColor": "#F4F4F4",
                                "borderBottomWidth": 0.5,
                                "borderColor": "#cccccc",
                                "paddingHorizontal": 15,
                              },
                              undefined,
                            ]
          }
>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                      }
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Array [
                                                  Object {
                                                    "color": "#000",
                                                    "fontSize": 16,
                                                    "marginRight": 15,
                                                    "paddingVertical": 14,
                                                  },
                                                  undefined,
                                                  false,
                                                ]
                              }
                    >
                              label
                    </Text>
                    <View
                              style={
                                        Array [
                                                  Object {
                                                    "flex": 1,
                                                  },
                                                  undefined,
                                                ]
                              }
                    >
                              <TextInput
                                        onChangeText={undefined}
                                        style={
                                                  Array [
                                                            Object {
                                                              "flex": 1,
                                                              "fontSize": 16,
                                                              "textAlign": "right",
                                                            },
                                                            undefined,
                                                            false,
                                                          ]
                                        }
                              />
                    </View>
          </View>
</View>,
        "_debugID": 4,
        "_renderedOutput": <View
          style={
                    Array [
                              Object {
                                "backgroundColor": "#F4F4F4",
                                "borderBottomWidth": 0.5,
                                "borderColor": "#cccccc",
                                "paddingHorizontal": 15,
                              },
                              undefined,
                            ]
          }
>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                      }
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Array [
                                                  Object {
                                                    "color": "#000",
                                                    "fontSize": 16,
                                                    "marginRight": 15,
                                                    "paddingVertical": 14,
                                                  },
                                                  undefined,
                                                  false,
                                                ]
                              }
                    >
                              label
                    </Text>
                    <View
                              style={
                                        Array [
                                                  Object {
                                                    "flex": 1,
                                                  },
                                                  undefined,
                                                ]
                              }
                    >
                              <TextInput
                                        onChangeText={undefined}
                                        style={
                                                  Array [
                                                            Object {
                                                              "flex": 1,
                                                              "fontSize": 16,
                                                              "textAlign": "right",
                                                            },
                                                            undefined,
                                                            false,
                                                          ]
                                        }
                              />
                    </View>
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InlineLabelTextInput
    input={Object {}}
    label="label"
    meta={
        Object {
            "error": null,
            "invalid": false,
            "touched": false,
          }
    }
/>,
}
`;

exports[`InlineLabelTextInputField component renders correctly with error 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Array [
            Object {
              "backgroundColor": "#F4F4F4",
              "borderBottomWidth": 0.5,
              "borderColor": "#cccccc",
              "paddingHorizontal": 15,
            },
            undefined,
          ]
    }
>
    <View
        style={
            Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
        }
    >
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Array [
                    Object {
                      "color": "#000",
                      "fontSize": 16,
                      "marginRight": 15,
                      "paddingVertical": 14,
                    },
                    undefined,
                    Object {
                      "color": "rgb(243, 108, 96)",
                    },
                  ]
            }
        >
            label
        </Text>
        <View
            style={
                Array [
                    Object {
                      "flex": 1,
                    },
                    undefined,
                  ]
            }
        >
            <TextInput
                onChangeText={undefined}
                style={
                    Array [
                        Object {
                          "flex": 1,
                          "fontSize": 16,
                          "textAlign": "right",
                        },
                        undefined,
                        Object {
                          "color": "rgb(243, 108, 96)",
                        },
                      ]
                }
            />
        </View>
    </View>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
            Object {
                "color": "rgb(243, 108, 96)",
                "fontSize": 16,
                "marginBottom": 10,
              }
        }
    >
        error
    </Text>
</View>,
  "nodes": Array [
    <View
      style={
            Array [
                  Object {
                    "backgroundColor": "#F4F4F4",
                    "borderBottomWidth": 0.5,
                    "borderColor": "#cccccc",
                    "paddingHorizontal": 15,
                  },
                  undefined,
                ]
      }
>
      <View
            style={
                  Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
            }
      >
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Array [
                              Object {
                                "color": "#000",
                                "fontSize": 16,
                                "marginRight": 15,
                                "paddingVertical": 14,
                              },
                              undefined,
                              Object {
                                "color": "rgb(243, 108, 96)",
                              },
                            ]
                  }
            >
                  label
            </Text>
            <View
                  style={
                        Array [
                              Object {
                                "flex": 1,
                              },
                              undefined,
                            ]
                  }
            >
                  <TextInput
                        onChangeText={undefined}
                        style={
                              Array [
                                    Object {
                                      "flex": 1,
                                      "fontSize": 16,
                                      "textAlign": "right",
                                    },
                                    undefined,
                                    Object {
                                      "color": "rgb(243, 108, 96)",
                                    },
                                  ]
                        }
                  />
            </View>
      </View>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                  Object {
                        "color": "rgb(243, 108, 96)",
                        "fontSize": 16,
                        "marginBottom": 10,
                      }
            }
      >
            error
      </Text>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <InlineLabelTextInput
        input={Object {}}
        label="label"
        meta={
                Object {
                        "error": "error",
                        "invalid": true,
                        "touched": true,
                      }
        }
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "input": Object {},
          "label": "label",
          "meta": Object {
            "error": "error",
            "invalid": true,
            "touched": true,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Array [
                              Object {
                                "backgroundColor": "#F4F4F4",
                                "borderBottomWidth": 0.5,
                                "borderColor": "#cccccc",
                                "paddingHorizontal": 15,
                              },
                              undefined,
                            ]
          }
>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                      }
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Array [
                                                  Object {
                                                    "color": "#000",
                                                    "fontSize": 16,
                                                    "marginRight": 15,
                                                    "paddingVertical": 14,
                                                  },
                                                  undefined,
                                                  Object {
                                                    "color": "rgb(243, 108, 96)",
                                                  },
                                                ]
                              }
                    >
                              label
                    </Text>
                    <View
                              style={
                                        Array [
                                                  Object {
                                                    "flex": 1,
                                                  },
                                                  undefined,
                                                ]
                              }
                    >
                              <TextInput
                                        onChangeText={undefined}
                                        style={
                                                  Array [
                                                            Object {
                                                              "flex": 1,
                                                              "fontSize": 16,
                                                              "textAlign": "right",
                                                            },
                                                            undefined,
                                                            Object {
                                                              "color": "rgb(243, 108, 96)",
                                                            },
                                                          ]
                                        }
                              />
                    </View>
          </View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "rgb(243, 108, 96)",
                                        "fontSize": 16,
                                        "marginBottom": 10,
                                      }
                    }
          >
                    error
          </Text>
</View>,
        "_debugID": 6,
        "_renderedOutput": <View
          style={
                    Array [
                              Object {
                                "backgroundColor": "#F4F4F4",
                                "borderBottomWidth": 0.5,
                                "borderColor": "#cccccc",
                                "paddingHorizontal": 15,
                              },
                              undefined,
                            ]
          }
>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                      }
                    }
          >
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Array [
                                                  Object {
                                                    "color": "#000",
                                                    "fontSize": 16,
                                                    "marginRight": 15,
                                                    "paddingVertical": 14,
                                                  },
                                                  undefined,
                                                  Object {
                                                    "color": "rgb(243, 108, 96)",
                                                  },
                                                ]
                              }
                    >
                              label
                    </Text>
                    <View
                              style={
                                        Array [
                                                  Object {
                                                    "flex": 1,
                                                  },
                                                  undefined,
                                                ]
                              }
                    >
                              <TextInput
                                        onChangeText={undefined}
                                        style={
                                                  Array [
                                                            Object {
                                                              "flex": 1,
                                                              "fontSize": 16,
                                                              "textAlign": "right",
                                                            },
                                                            undefined,
                                                            Object {
                                                              "color": "rgb(243, 108, 96)",
                                                            },
                                                          ]
                                        }
                              />
                    </View>
          </View>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                              Object {
                                        "color": "rgb(243, 108, 96)",
                                        "fontSize": 16,
                                        "marginBottom": 10,
                                      }
                    }
          >
                    error
          </Text>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InlineLabelTextInput
    input={Object {}}
    label="label"
    meta={
        Object {
            "error": "error",
            "invalid": true,
            "touched": true,
          }
    }
/>,
}
`;

exports[`InlineLabelTextInputField field renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Field
    component={[Function]}
    name="Test_Input_Name"
/>,
  "nodes": Array [
    <Field
      component={[Function]}
      name="Test_Input_Name"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <InlineLabelTextInputField
        textInputName="Test_Input_Name"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "textInputName": "Test_Input_Name",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Field
          component={[Function]}
          name="Test_Input_Name"
/>,
        "_debugID": 2,
        "_renderedOutput": <Field
          component={[Function]}
          name="Test_Input_Name"
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <InlineLabelTextInputField
    textInputName="Test_Input_Name"
/>,
}
`;
